gcc/rust/ast/rust-pattern.h

```diff
@@ -91,6 +91,15 @@ class IdentifierPattern : public Pattern
       is_mut (is_mut), to_bind (std::move (to_bind)), locus (locus)
   {}
 
+  IdentifierPattern (NodeId node_id, Identifier ident, Location locus,
+		     bool is_ref = false, bool is_mut = false,
+		     std::unique_ptr<Pattern> to_bind = nullptr)
+    : Pattern (), variable_ident (std::move (ident)), is_ref (is_ref),
+      is_mut (is_mut), to_bind (std::move (to_bind)), locus (locus)
+  {
+    this->node_id = node_id;
```

Comment:
Wouldn't it be better to construct the node_id in the constructor list rather than the body?

Reply:
Good find will fix.